### PR TITLE
refactor(modeling)!: V3 convert poly2 vertices to points

### PR DESCRIPTION
**Pull request by platypii**
This PR changes `poly2.vertices` to `poly2.points`.

The problems is that otherwise there is no difference between the structure of poly2 and poly3. So you would have the following weirdness:

```js
poly2.isA( poly3.create() ) == true
poly3.isA( poly2.create() ) == true
```

By using `points` instead of `vertices` we can tell the difference between a poly2 and a poly3.

More generally, I like the idea of using "points" to refer to 2D points, and "vertices" to refer to 3D points. I think it could help make JSCAD code easier to understand if we are consistent about that.

I also added tests for `poly2.reverse` and `poly2.toPoints` since I noticed that they were not covered by existing tests.

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Does your submission pass tests?

### CODE REVIEW

File: packages/modeling/src/operations/extrusions/earcut/assignHoles.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import area from '../../../maths/utils/area.js'
-import toOutlines from '../../../geometries/geom2/toOutlines.js'
-import arePointsInside from '../../../geometries/poly2/arePointsInside.js'
+import { toOutlines } from '../../../geometries/geom2/index.js'
```
Comment (z3dev) [inline packages/modeling/src/operations/extrusions/earcut/assignHoles.js:2]: In general, internal logic should import functions directly, which prevents circular dependencies. (It happens)

why the change?

**Review by z3dev — CHANGES_REQUESTED** [review]
@platypii just one little nit.

File: packages/modeling/src/operations/extrusions/earcut/assignHoles.js (hunk 1 above)
Comment (platypii) [reply packages/modeling/src/operations/extrusions/earcut/assignHoles.js:2]: I was trying to be more consistent with what I have seen elsewhere in JSCAD. It seems like the preferred pattern is to use destructuring to import from index.js files. This seems better because it is using the JSCAD standard API, not making assumptions about where files are located. I can revert this change if you prefer though.

Comment (platypii) [reply packages/modeling/src/operations/extrusions/earcut/assignHoles.js:2]: If you look at existing imports of geometry functions inside modeling, almost all of them import from index.js. In fact `assignHoles.js` is basically the only place that imported directly. That's why I changed it.

```bash
grep -rn import.*from.*geometries packages/modeling/src
```

Comment (z3dev) [reply packages/modeling/src/operations/extrusions/earcut/assignHoles.js:2]: Yup. My mistake.

It's the general index that internals should not use, i.e. geometries/index.js, etc.

**Review by z3dev — APPROVED** [review]
@platypii do you ever sleep? :)
